### PR TITLE
Add Result to public fn return types

### PR DESCRIPTION
**Pull request by jayvdb**
This will cause conflicts with https://github.com/kurotych/sqlant/pull/27.

Once these are in, the other unwraps can be solved.

### CODE REVIEW

File: src/mermaid_generator.rs [hunk 1]
```diff
@@ -145,12 +143,16 @@ impl<'a> MermaidGenerator<'a> {
...
+        mut sql_erd: SqlERData,
+        opts: &GeneratorConfigOptions,
+    ) -> Result<String, crate::SqlantError> {
         Self::preprocess(&mut sql_erd);
         let entities: Vec<String> = sql_erd
             .tables
             .iter()
-            .map(|tbl| self.entity_render(tbl, opts))
+            .map(|tbl| self.entity_render(tbl, opts).unwrap())
             .collect();
```
Comment (kurotych) [inline src/mermaid_generator.rs:156]:
```rust
let entities: Vec<String> = sql_erd
    .tables
    .iter()
    .map(|tbl| self.entity_render(tbl, opts))
    .collect::<Result<Vec<String>, crate::SqlantError>>()?;

File: src/plantuml_generator.rs [hunk 2]
```diff
@@ -114,26 +118,28 @@ impl<'a> PlantUmlDefaultGenerator<'a> {
...
+    fn generate(
+        &self,
+        sql_erd: SqlERData,
+        opts: &GeneratorConfigOptions,
+    ) -> Result<String, crate::SqlantError> {
         let entities: Vec<String> = sql_erd
             .tables
             .iter()
-            .map(|tbl| self.entity_render(tbl, opts))
+            .map(|tbl| self.entity_render(tbl, opts).unwrap())
```
Comment (kurotych) [inline src/plantuml_generator.rs:142]: We can use "?" see message above

**Review by kurotych — COMMENTED** [review]
I think we can remove two more "unwraps()" (left comments)
Also please fix merge conflicts and we are good to go

Thank you for your contribution!

File: src/plantuml_generator.rs (hunk 2 above)
Comment (jayvdb) [reply src/plantuml_generator.rs:142]: There are a lot more unwraps & panics which can be removed.
This PR is about adding `Result` to the public APIs.

**Comment by jayvdb** [comment]
Feel free to made those changes you suggested.  I wont have time to do any more today.

**Pull request merged**